docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+}
+</style>
+* Table of Contents
+{:toc}
 
-## Design & implementation
+## How To Use This Guide
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+This **Developer Guide** aims to get developers familiarised with the design and implementation of **CLIrcuit Assistant**. The following table indicates the symbols used to aid the understanding of the guide. This guide also assumes that the reader has basic understanding of *UML Diagrams*. [To access the **User Guide** instead, click here.](UserGuide.md)
 
+| Symbol/Format | Meaning |
+|:---------------:|:--------|
+|:information_source:|An informational source.|
+|:exclamation:|A warning.|
+|**Bolded**|A keyword.|
+|*Italicised*|Technical word.|
+|[Hyperlinked](#)|Leads to the appropriate section.|
+|`Code`|Text that appears on the CLI / in code.|
 
-## Product scope
-### Target user profile
+--------------------------------------------------------------------------------------------------------------------
 
-{Describe the target user profile}
+## Setting up, getting started
+Refer to the guide [*Setting up and getting started*](SettingUp.md).
 
-### Value proposition
+--------------------------------------------------------------------------------------------------------------------
 
-{Describe the value proposition: what problem does it solve?}
+## Design Architecture (Wira) <a name="design"></a>
 
-## User Stories
+![ArchitectureDiagram](diagrams/ArchitectureDiagram.png)
+
+<small><i>Figure 1</i></small>
+
+The **Architecture Design** given above explains the high-level design of the App. Given below is a quick overview of each component.
+
+**`Duke`** is the main class of the application, and handles the app launch, initializing the appropriate classes to be used.
+
+The rest of the app consists of three components.
+
+* [**`UI`**](#ui-component): The UI of the App.
+* [**`Logic`**](#logic-component): The command executor.
+* [**`Model`**](#model-component): Holds the data of the App in memory.
+
+### UI component
+
+![UiDiagram](diagrams/UiClassDiagram.png)
+
+<small><i>Figure 2</i></small>
+
+The `Ui` component
+
+* Reads user input as lines using `readLine()`.
+* Prints every `Command` object that is parsed using `printMessage()`.
+* Prints any user input error that might occur using `showError()`.
+
+![UiSequence](diagrams/UiSequenceDiagram.png)
+
+<small><i>Figure 3</i></small>
+
+### Logic component
+
+![LogicDiagram](diagrams/LogicClassDiagram.png)
+
+<small><i>Figure 4</i></small>
+
+1. `Logic` stores a current `Template` object in [`Model`](#model-component) that represents the current circuit configuration.
+1. `Logic` uses the `Parser` class to parse the user command.
+1. This results in a `Command` object which is executed in `Duke`.
+1. The command execution can affect the `Model` (e.g. setting a value).
+1. In addition, the `Ui` may also perform certain actions, such as displaying help to the user.
```

Comment:
Would it be good to include the `Ui` component (similar to `Model`) in the above diagram for the `Logic` component, to keep more in line with the description that follows?

Reply:
Line removed in https://github.com/AY2021S1-CS2113T-W13-3/tp/pull/121.